pyspecProcScripts/phasing.py

```diff
@@ -222,6 +262,21 @@ def filter_ranges(B, A):
                 any(b[0] <= a[0] and b[1] >= a[1] for a in A)]
     peakrange = filter_ranges(wide_ranges, narrow_ranges)
     assert len(peakrange) == 1
+    if len(peakrange) > 1:
+        max_range_width = max([thisrange[1]-thisrange[0] for thisrange in peakrange])
+        range_gaps = [peakrange[j+1][0] - peakrange[j][1] for j in range(len(peakrange)-1)]
+        # {{{ if the gaps are all smaller than the max peak that was found, we
+        #     just have "breaks" in the peak, so merge them.  Otherwise, fail.
+        if any(np.array(range_gaps) > max_range_width):
+            if fl is not None:
+                fl.next("debug filter ranges")
+                fl.plot(freq_envelope, human_units=False)
+                for thisrange in peakrange:
+                    fl.plot(freq_envelope[direct:thisrange], human_units=False)
+            raise ValueError("finding more than one peak!")
+        else:
+            peakrange = [(peakrange[0][0],peakrange[-1][1])]
+        # }}}
```

Comment:
handles when more than one peak is found by identifying "breaks" in the peak vs the actual large peak